src/common/network/NetworkUtils.cpp

```diff
@@ -215,15 +215,32 @@ std::string NetworkUtils::intToIPv4(IPv4 ip) {
     return buf;
 }
 
-StatusOr<HostAddr> NetworkUtils::toHostAddr(folly::StringPiece ip, int32_t port) {
+StatusOr<std::vector<HostAddr>> NetworkUtils::toHostAddr(folly::StringPiece ip, int32_t port) {
+    std::vector<HostAddr> addrs;
     IPv4 ipV4;
-    if (!ipv4ToInt(ip.toString(), ipV4)) {
-        return Status::Error("Bad ip format:%s", ip.start());
+    std::string str = ip.toString();
+    if (ipv4ToInt(str, ipV4)) {
+        addrs.emplace_back(std::move(ipV4), port);
+        return addrs;
     }
-    return std::make_pair(ipV4, port);
+
+    struct hostent* h;
+    h = gethostbyname(str.c_str());
```

Comment:
According to the man pages, `gethostbyname` is already obsolete, while the alternative interface is `getaddrinfo`.

Reply:
In some older systems does not provide getaddrinfo, so here we need to detect the system for getaddrinfo support.So I will rewrite it here.

Reply:
`getaddrinfo` was introduced since glibc 2.3.4, I don't think it's a problem for us.

Reply:
Ok, then I will ignore gethostbyname.